staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -235,6 +246,10 @@ func (c *dynamicResourceClient) Delete(ctx context.Context, name string, opts me
 }
 
 func (c *dynamicResourceClient) DeleteCollection(ctx context.Context, opts metav1.DeleteOptions, listOptions metav1.ListOptions) error {
+	if err := c.validateURLSegments(""); err != nil {
```

Comment:
This is a very confusing line if you haven't read the function and didn't expect that it was also validating the namespace :)

Reply:
Agreed. I could have done a free function that receives variable arguments.
```
func validateURLSegments(segment ...string) error
```

Reply:
It's probably not worth changing, and if you did I'd probably just change the name?